edu-services/cm-service/cm-impl/hibernate-impl/impl/src/java/org/sakaiproject/coursemanagement/impl/CourseManagementServiceHibernateImpl.java

```diff
@@ -471,13 +464,16 @@ public String getSectionCategoryDescription(String categoryCode) {
 	}
 
 	public String getEnrollmentStatusDescription(String statusId) {
-		return new ResourceLoader("enrollmentstatus").getString(statusId, statusId);
+		return enrollmentsMessages.getString(statusId, statusId);
 	}
 
 	public Map<String, String> getEnrollmentStatusDescriptions(Locale locale) {
-
-		ResourceLoader rl = new ResourceLoader("enrollmentstatus");
-		return ((Set<String>) rl.keySet()).stream().collect(Collectors.toMap(k -> k, k -> rl.getString(k, k)));
+		enrollmentsMessages.setContextLocale(locale);
```

Comment:
Nice, another bug fixes in the original code, unused locale.

Reply:
locale can be null there